website: First draft of "Refactoring" documentation, about "moved" blocks (#29126)
hashicorp/terraform: website/docs/language/modules/syntax.html.md

## Transferring Resource State Into Modules

> When refactoring an existing configuration to split resources across multiple
> child modules, just moving the `resource` blocks into the child modules would
> cause Terraform to see the new location as an entirely different resource, and
> so Terraform will propose to destroy all resource instances at the old address
> and create new instances at the new address.

Reviewer: ```suggestion
Moving `resource` blocks from one module into several child modules causes Terraform to see the new location as an entirely different resource. As a result, Terraform plans to destroy all resource instances at the old address and create new instances at the new address.
```

> To preserve existing objects when refactoring between modules, you can use
> [refactoring blocks](develop/refactoring.html) to record the old and new addresses
> for each change you made. Terraform will then treat existing objects at the
> old addresses as if they had originally been created at the corresponding
> new addresses.

Reviewer: ```suggestion
To preserve existing objects, you can use [refactoring blocks](develop/refactoring.html) to record the old and new addresses for each resource. This directs Terraform to treat existing objects at the
old addresses as if they had originally been created at the corresponding new addresses.
```

## Replacing resources within a module

> If you have an object that needs to be replaced with a new object for a reason
> that isn't automatically visible to Terraform, such as if a particular virtual
> machine is running on degraded underlying hardware, you can use
> [the `-replace=...` planning option](https://www.terraform.io/docs/cli/commands/plan.html#replace-address)
> to force Terraform to propose replacing that object.

Reviewer: ```suggestion
You may have an object that needs to be replaced with a new object for a reason
that isn't automatically visible to Terraform, such as if a particular virtual
machine is running on degraded underlying hardware. In this case, you can use
[the `-replace=...` planning option](https://www.terraform.io/docs/cli/commands/plan.html#replace-address)
to force Terraform to propose replacing that object.
```

> If the object in question belongs to a resource within a nested module,
> specify the full path to that resource including all of the nested module
> steps leading to it. For example:

Reviewer: ```suggestion
If the object belongs to a resource within a nested module,
specify the full path to that resource including all of the nested module
steps leading to it. For example:
```

```shellsession
$ terraform plan -replace=module.example.aws_instance.example
```

> The above would select a `resource "aws_instance" "example"` declared inside
> a `module "example"` child module declared inside your root module.

Reviewer: ```suggestion
The above selects a `resource "aws_instance" "example"` declared inside
a `module "example"` child module declared inside your root module.
```
